### DISCUSSION TITLE
The QoS 2 Message Receive Queue is not being cleared

### DISCUSSION THREAD

**Discussion by ValentinaDrozdova** [emqx/emqx#12664] [Q&A]
During the time specified in the Management > MQTT Settings > Session > Max Awaiting PUBREL Timeout settings, the queue is not cleared if 100 or more messages arrive at the broker. I've commented out the pubrel sending by the client and also set the session_expiry_interval and clean_session = False for the client, but it didn't help. Could you please advise if I'm doing everything correctly and if there are any other client parameters that need to be configured?
Max Awaiting PUBREL Timeout = 300s

**Comment by kjellwinblad** [comment]
Hi! Thanks for the question. The setting you are referring to (Max Awaiting PUBREL Timeout) has the following documentation:

> Specifies the amount of time to wait for a publish of a QoS 2 message with no PUBREL received. When this limit is reached, EMQX will release the packet ID and also output a warning level log. Note: EMQX’s forwarding of the received QoS 2 message is independent from the receiving of PUBREL

And the other setting that you are implicitly referring to (Max Awaiting PUBREL) has the following documentation: 

> Specifies the maximum number of pending QoS 2 messages in each session until either PUBREL is received or timed out. When this limit is reached, new QoS 2 PUBLISH requests will be rejected with error code 147(0x93).

The documentation does not mention anything about clearing any queue. It talks about dropping messages when the limit is reached, and dropping the packet ID and outputting a log level warning but not about clearing any queue. Could you try to clarify what you mean by "Receive Queue is not being cleared"?

**Reply by ValentinaDrozdova** [reply]
I apologize for the misunderstanding. I meant deleting messages. The issue is that messages are not being deleted upon reaching the limit, and no warnings or rejections with error code 147 are being displayed.

**Reply by YraganTron** [reply]
Hello! I have the same problem.
I'm interested in a situation where my client doesn't send PUBREL for some messages and after that my queue (maximum number of pending PUBREL) is full.

After this my client reconnects and is unable to send messages because my queue (maximum number of pending PUBRELs) is full.
My client has a session_expiry_interval property because it is important for the client to receive messages.
After reconnecting, my client is not aware of the previous messages.
I'm looking for a way to allow my client to send messages after reconnecting. As I understand it, this requires that the queue (the maximum number of waiting PUBRELs) stops being full.
